soc/arm/microchip_mec/mec1501/device_power.c

```diff
@@ -194,23 +191,128 @@ static void deep_sleep_restore_timers(void)
 
 	p = &ds_timer_tbl[0];
 	for (i = 0; i < NUM_DS_TIMER_ENTRIES; i++) {
-		REG32(p->addr) = timers[i] & ~p->restore_mask;
+		REG32(p->addr) = ds_ctx.timers[i] & ~p->restore_mask;
 		p++;
 	}
 }
 
+#ifdef DEEP_SLEEP_PERIPH_SAVE_RESTORE_EXTENDED
+
+static void deep_sleep_save_blocks(void)
+{
+	/* ADC Power saving feature is enabled */
+	ds_ctx.adc_info.adc_ctrl = ADC_REGS->CONTROL;
```

Comment:
@scottwcpg does the new extended save & restore looks correct?

Reply:
The SLEEP_ALL bit in PCR SYS_SLP_CTRL forces all peripheral SLP_EN to 1 when WFI triggers the PCR HW.  Why do you need to manually set and clear PCR SLP_EN bits?
The code is confusing because most of it appears to be for reducing power in deep sleep state vs requirement to enter deep sleep state. More comments indicating which chunks are for power saving and which ones are required to cause the peripheral to clear its CLK_REQ would be helpful for future maintainers.
Timer table: 
Basic timers have a HALT bit when set stops the counter at its current count and de-asserts CLK_REQ. This is better than save/restore of whole control register.
ADC: Why clear ADC CONFIG?  All that's needed is clearing ACTIVATE bit in CONTROL.  If a conversion is in progress the ADC will stop when the conversion is done.
Are there hard requirements the chip must enter the lowest power state with a very short time?
I assume you have no requirements to wake from an external I2C controller since the I2C blocks are being disabled?

Reply:
Yes, all this is in the context of power saving since we don't have yet MEC15 drivers that are power aware. (No CONFIG_DEVICE_MANAGEMENT support )

If that would have been in place none of this is required. As discuss offline MCHP has no ETA for that happen hence this PR.

1) Will remove the incorrect SLP_EN settings, I misinterpreted your comment about UART as only not needed for UART.

2) I documented these are for power savings it in the header 
What are you have a preference for a macro or Kconfig?    Any name i.e do you want to call this CONFIG_SOC_HW_BLOCK_OPTIMIZATION

/* Power optimization if certain HW blocks are not used.
 * These are not part of any Zephyr subsystem.
 *  #define DEEP_SLEEP_PERIPH_SAVE_RESTORE_EXTENDED

3) Timer are your original implementation just clean up a bit, are you now saying that has always been incorrect?

4) Regarding ADC this is recommendation I got from MCHP HW designer, please double check.

5) Currently I2C slave wake is not considered, please point exactly what is been masked

Reply:
If UART save/restore as implemented is acceptable then no change.  MCHP will investigate different new methods for MEC172x.
Timer save/restore works. MCHP will use new method in next SoC MEC172x.
I didn't know ADC recommendation came from a HW designer(needs data sheet update).  Ignore my comments. 
I2C not use for wake then no issue since I2C should be idle in sleep path.